src/open_samus_returns_rando/specific_patches/heat_room_patches.py

```diff
@@ -14,226 +22,87 @@ def _get_new_logic_shape(editor: PatcherEditor, points: int) -> Container:
-    ht_to_copy = {"scenario": "s010_area1", "layer": "2", "actor": "TG_Heat_001"}
-    heat_actor = editor.resolve_actor_reference(ht_to_copy)
-    return heat_actor
-
-
-def _patch_area_2b(editor: PatcherEditor):
-    # area2b => shape is already correct but ms forgot to add one room
-    heat_trigger_2b = {"scenario": "s025_area2b", "layer": "2", "actor": "TG_SP_Heat_001"}
-    scenario_2b = editor.get_scenario(heat_trigger_2b["scenario"])
-    scenario_2b.add_actor_to_entity_groups("collision_camera013", heat_trigger_2b["actor"])
-
-def _patch_area_3b(editor: PatcherEditor):
-    name_of_scenario = "s033_area3b"
-
-    scenario_3b = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([-500.0, 3800.0, 3900.0, 6700.0])
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
-
-    points[0]["x"] = -500.0
-    points[0]["y"] = 3800.0
-    points[1]["x"] = 3900.0
-    points[1]["y"] = 3800.0
-    points[2]["x"] = 3900.0
-    points[2]["y"] = 6700.0
-    points[3]["x"] = -500.0
-    points[3]["y"] = 6700.0
-
-    scenario_3b.raw["objects_c"]["LS_Heat_001"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    editor.copy_actor(name_of_scenario, (1000.0, 5000.0, 0.0), heat_actor, "TG_Heat_003", 2)
-    scenario_3b.add_actor_to_entity_groups("collision_camera_036", "TG_Heat_003")
-
-def _patch_area_3c(editor: PatcherEditor):
-    name_of_scenario = "s036_area3c"
-
-    scenario_3c = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([10800.0, 1700.0, 21700.0, 5800.0])
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
-
-    points[0]["x"] = 10800.0
-    points[0]["y"] = 1700.0
-    points[1]["x"] = 21700.0
-    points[1]["y"] = 1700.0
-    points[2]["x"] = 21700.0
-    points[2]["y"] = 5800.0
-    points[3]["x"] = 10800.0
-    points[3]["y"] = 5800.0
-
-    scenario_3c.raw["objects_c"]["LS_Heat_001"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    editor.copy_actor(name_of_scenario, (19269.0, 3669.0, 0.0), heat_actor, "TG_Heat_003", 2)
-
-    scenario_3c.add_actor_to_entity_groups("collision_camera_022", "TG_Heat_003")
-    scenario_3c.add_actor_to_entity_groups("collision_camera_026", "TG_Heat_003")
-
-def _patch_area_4(editor: PatcherEditor):
-    name_of_scenario = "s040_area4"
 
-    scenario_4 = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([-850.0, 300.0, 4250.0, 1950.0])
+class NewHeatActor(typing.NamedTuple):
+    scenario: str
+    position: list[float]
+    total_boundings: list[float]
+    points: list[float]
```

Comment:
The type is wrong. The values used in `new_heat_actors` for that field are at the moment of type `tuple[list[float],list[float],list[float],list[float]]`.
I think changing the type and values in `new_heat_actors` to `points: list[tuple[float, float]]` would be better.

With the other change, we then could add non recantgular shapes for whatever reason.